### PR TITLE
test: add more tests to topology spread constraints plugin (again)

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR adds more tests to the topology spread constraints plugin.

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

N/A

### Special notes for your reviewer

A minor part of this PR overlaps with another test PR.

### CODE REVIEW

File: pkg/scheduler/framework/score_test.go [hunk 1]
```diff
@@ -75,6 +75,22 @@ func TestClusterScoreEqual(t *testing.T) {
...
+		},
+		{
+			name: "s2 is nil",
+			s1: &ClusterScore{
+				TopologySpreadScore:   2,
+				AffinityScore:         5,
+				BoundOrScheduledScore: 1,
+			},
+		},
 	}
```
Comment (zhiying-lin) [inline pkg/scheduler/framework/score_test.go:94]: add a test when s1 & s2 are nil?

Comment (michaelawyu) [reply pkg/scheduler/framework/score_test.go:98]: Done.

**Review by zhiying-lin — APPROVED** [review]
